src/store/ducks/opensrp/events/utils.ts

```diff
@@ -0,0 +1,180 @@
+  version: number;
+  teamId: UUID;
+  _id: UUID;
+  _rev: string;
+}
+
+/** describes the foci information property */
+interface FociInformation {
+  id: UUID;
+  classification: string;
+  name: string;
+}
+
+/** describes the case information property for a n extracted event */
+export interface CaseInformation {
+  caseClassification: string;
+  diagnosisDate: DateString;
+  species: string;
+  notificationDate: DateString;
+  investigationDate: DateString;
+  houseNumber: string;
+  patientName: string;
+  surname: string;
+  age: string;
+}
+
+/** describes the event object after extracting it from rawEvent */
+export interface Event {
+  id: UUID;
+  baseEntityId: UUID;
+  caseNumber: string;
+  caseInformation: CaseInformation;
+  fociInformation: FociInformation;
+  fociInformationTitle: string;
+}
+
+/** lookup table to decide the title of foci information based on events flag details */
+const FociTitleLookup: Dictionary<string> = {
+  Site: FOCI_OF_RESIDENCE,
+  Source: FOCI_OF_INFECTION,
+};
+
+/** Takes the iso formated date and displays a readable date format
+ * @param {DateString} dateString - the ISO formated string
+ *
+ * @return {DateString} - a human friendly related date
+ */
+export const friendlyDate = (dateString: DateString): string => {
+  return moment(dateString).format('LLL');
+};
+
+/** extract the event details we require from what we receive from the api
+ * @param {RawEvent} rawEvent - the event object from the api
+ * @return {Event} - event object with fields
+ */
+export const extractEvent = (rawEvent: RawEvent): Event => ({
+  baseEntityId: rawEvent.baseEntityId,
+  caseInformation: {
+    age: rawEvent.details.age,
+    caseClassification: rawEvent.details.case_classification,
+    diagnosisDate: friendlyDate(rawEvent.details.ep3_create_date),
+    houseNumber: rawEvent.details.house_number,
+    investigationDate: friendlyDate(rawEvent.details.investigtion_date),
+    notificationDate: friendlyDate(rawEvent.details.ep1_create_date),
+    patientName: rawEvent.details.first_name,
+    species: rawEvent.details.species,
+    surname: rawEvent.details.surname,
+  },
+  caseNumber: rawEvent.details.case_number,
+  fociInformation: {
+    classification: rawEvent.details.focus_status,
+    id: rawEvent.details.focus_id,
+    name: rawEvent.details.focus_name,
+  },
+  fociInformationTitle: FociTitleLookup[rawEvent.details.flag],
+  id: rawEvent._id,
+});
+
+/** extracts events in batch
+ * @param {RawEvents} rawEvent - the event objects from the api
+ * @return {Event} - events object with fields
+ */
+export const extractEvents = (rawEvents: RawEvent[]): Event[] => {
+  return rawEvents.map<Event>(rawEvent => extractEvent(rawEvent));
+};
+
+/** describes the form an event will take after translating its properties */
+interface TranslatedEvent {
+  baseEntityId: UUID;
+  caseInformation: Dictionary<string>;
+  caseNumber: string;
+  fociInformation: Dictionary<string>;
+  fociInformationTitle: string;
+}
```

Comment:
What's the difference between `Event` and `TranslatedEvent`? Can one extend another?

Reply:
@moshthepitt @cKellyDesign The event is store in store and extracted from the rawEvent, howver the translatedEvent will be used during display , such that i can iterate over its keys and values display them without needing to worry about localizing each property name at render time

Reply:
@p-netm ok cool, maybe we can just add `for display` somewhere in the doc-string?

Reply:
ok

Reply:
@p-netm ok cool, maybe we can just add `for display` somewhere in the doc-string?